stetho/src/main/java/com/facebook/stetho/inspector/elements/ObjectDescriptor.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class ObjectDescriptor extends Descriptor {
+  private final Map<Class<?>, String> mClassToFriendlyNameCache =
```

Comment:
Ahh, that makes sense. I must've mentally slipped back into C#-mode where every string is its own allocation without any sharing (I can explain why it's done that way if you ever want to know, btw).